pandas/core/nanops.py

```diff
@@ -462,6 +461,14 @@ def nanmean(values, axis=None, skipna=True, mask=None):
     elif is_float_dtype(dtype):
         dtype_sum = dtype
         dtype_count = dtype
+    elif is_datetime64_dtype(dtype) or is_datetime64tz_dtype(dtype):
+        from pandas import DatetimeIndex
+        masked_vals = values
+        if mask is not None:
+            masked_vals = values[~mask]
+        the_mean = DatetimeIndex(masked_vals).mean(skipna=skipna)
```

Comment:
pls try to follow the patterns in this module
_wrap_results exists for a reason

again i see lots of reinventing the wheel